JavaScript/JavaScriptSDK/Logging.ts

```diff
@@ -81,5 +83,48 @@
                 }
             }
         }
+        
+        /**
+         * Resetting the throttle limits for Internal events
+         */
+        public static resetInternalEventsThrottle(): void {
+            this._eventCount = 0;
+        }
+
+        /**
+         * Sets the limit for the number of internal events before they are throttled
+         */
+        public static setMaxAllowedInternalThrottleLimit(limit: number): void {
+            if (!limit) {
```

Comment:
If the `_InternalLogging` class is not supposed to be used by the customers of our API, it would be important to follow the [fail fast](http://www.martinfowler.com/ieeeSoftware/failFast.pdf) principle and avoid the [defensive coding anti-pattern](http://blog.iprofs.nl/2011/03/28/defensive-programming-as-anti-pattern/).